doc/multi_asic/SONiC_multi_asic_hld.md

```diff
@@ -0,0 +1,937 @@
+ - Route reflector is configured on the backend ASICs
+ - All the ASICs have the same view of the network.
+
+![BGP on backend ASICs](images/bgp_on_backend_asic.jpg)
+
+ In the above picture,
+
+  - ASIC0 and ASIC1 have ibgp sessions established with the backend ASIC, ASIC2.
+  - Router A and Router B establish eBGP sessions with ASIC0 and ASIC1 respectively
+  - Let say a prefix, _192.168.199.193_, is learnt from Router A on ASIC0.
+  ```
+    root@sonic:/home/admin# ip -n asic0 route show  192.168.199.193
+    192.168.199.193 proto 186 src 172.16.132.64 metric 20
+        nexthop via 10.106.0.1  dev PortChannel1002 weight 1
+  ```
+  - ASIC0 advertises this prefix to ASIC2. On ASIC2 the nexthop is the interface connected to ASIC0
+```
+    root@sonic:/home/admin# ip -n asic2 route show  192.168.199.193
+    192.168.199.193 proto 186 src 172.16.132.64 metric 20
+        nexthop via 10.0.107.1  dev PortChannel4009 weight 1
+
+```
+- ASIC1 learns these routes from ASIC2 as route reflection is configured on ASIC2
+```
+    root@sonic:/home/admin# ip -n asic1 route show  192.168.199.193
+    192.168.199.193 proto 186 src 172.16.132.64 metric 20
+        nexthop via 10.0.107.12  dev PortChannel4007 weight 1
+```
+  - A packet with destination ip as 192.168.199.193
+      - ingresses the device on ASIC1 and gets routed to ASIC2.
+      - ASIC2 then routes the packet to ASIC0.
+      - on ASIC0 it is routed and egresses out of the device.
+
+  - In this approach, since the packets can traverse through multiple internal hops, the ttl decrement can be more than 1.
+
+#### 2.3.2.2. VLAN/Cross-connect on backend ASICs
+ In this method:
+ - BGP is running only on the Frontend ASICs.
+ - The backend ASICs will switch packets using vlans or vlan-crossconnect.
+ - BGP sessions are formed between Frontend ASICs to form a mesh
+ - A unique VLAN is used for communication between a pair of Frontend ASIC
+
+ ![Vlan on backend ASICs](images/vlan_cc.jpg)
+
+ In the above diagram,
+  - Vlan 10 is used for communication between the Frontend ASICs, ASIC0 and ASIC1.
+    
+    - The ip address are configured on the Vlan as shown in the diagram.
+  -  The backend ASIC, ASIC4, can be configured in two ways
+    - Configure Vlan 10 and add ports connecting ASIC0 and ASIC1 as tagged members
+    - Configure Vlan cross-connect entries to switch packets from ASIC0 to ASIC1 and vice-versa.
+  - ASIC1 and ASIC0 have a iBGP session established
+  - The prefixes from an external Router are learnt on ASIC0  and are advertised to ASIC1 and vice-versa
+  - A packet from external Routers ingressing on ASIC0 and egressing on ASIC1 is routed twice
+  - For a device with N Frontend ASICs and M Backend ASICs, the number of BGP sessions and the number of vlans used for internal communications will be
+
+    ```
+    (( N * (N-1))/2) * M                                                                               
+    ```
+#### 2.3.2.3. Comparison of both approaches

```

Comment:
Which proposal does this design consider to implemement? or do you provide both?